src/Models/UserIdentityModel.php

```diff
@@ -25,6 +26,146 @@ class UserIdentityModel extends Model
     ];
     protected $useTimestamps = true;
 
+    public function getAccessTokenByToken(string $token): ?AccessToken
+    {
+        return $this
+            ->where('type', 'access_token')
+            ->where('secret', hash('sha256', $token))
+            ->asObject(AccessToken::class)
+            ->first();
+    }
+
+    /**
+     * @param int|string $userId
+     */
+    public function getAccessToken($userId, string $token): ?AccessToken
+    {
+        return $this->where('user_id', $userId)
+            ->where('type', 'access_token')
+            ->where('secret', hash('sha256', $token))
+            ->asObject(AccessToken::class)
+            ->first();
+    }
+
+    /**
+     * Given the ID, returns the given access token.
+     *
+     * @param int|string $id
+     * @param int|string $userId
+     */
+    public function getAccessTokenById($id, $userId): ?AccessToken
+    {
+        return $this->where('user_id', $userId)
+            ->where('type', 'access_token')
+            ->where('id', $id)
+            ->asObject(AccessToken::class)
+            ->first();
+    }
+
+    /**
+     * @param int|string $userId
+     *
+     * @return AccessToken[]
+     */
+    public function getAllAccessTokens($userId): array
+    {
+        return $this
+            ->where('user_id', $userId)
+            ->where('type', 'access_token')
+            ->asObject(AccessToken::class)
+            ->find();
```

Comment:
```suggestion
            ->findAll();
```
I realize both work but I want to move away from `find()` as a return type of single/multiple results.

Reply:
Done.